public/app/engagementindex/leaderboard/leaderboardFactory.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2015 UC Berkeley (UCB) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+(function(angular) {
+
+  'use strict';
+
+  angular.module('collabosphere').factory('leaderboardFactory', function(utilService, $http) {
```

Comment:
Yeah, either would work I think. I've tried to follow the recommendations at https://scotch.io/tutorials/angularjs-best-practices-directory-structure and split everything up into components. However, their examples don't have any factories, and maybe splitting the factories per component is taking it too far.

Would you like me to take the factories out of the components, consolidate them and put them higher up in the structure?